src/racing/actors.c

```diff
@@ -45,18 +45,20 @@ void cleanup_red_and_green_shells(struct ShellActor *shell) {
     s32 actorIndex;
     struct ShellActor *compare;
 
+    // try finding the dead green shell
     for (actorIndex = gNumPermanentActors; actorIndex < ACTOR_LIST_SIZE; actorIndex++) {
         compare = (struct ShellActor *)&gActorList[actorIndex];
         if ((shell != compare) && !(compare->flags & ACTOR_IS_NOT_EXPIRED) && (compare->type == ACTOR_GREEN_SHELL)) {
             if (compare->state == MOVING_SHELL) {
-                func_8000EE58(actorIndex);
+                remove_actor_in_unexpired_actor_list(actorIndex);
             }
-            gNumSpawnedShells -= 1;
+            gNumSpawnedShells--;
```

Comment:
I'm fine with this.

Just keep-in-mind that `obj--` does not necessarily mean `obj -= 1` because if it's a ptr or array it will be `obj -= sizeof(Obj)` or `obj -= 0x10` which means programmers sometimes prefer `-= 1` because it's clear that it's always -= 1 and not anything else.

However, I'm don't have any issue with `--` just wanted to mention that.

Reply:
ok ok